release-notes/8.0/preview/api-diff/preview4/Microsoft.NETCore.App/8.0-preview4_Microsoft.VisualBasic.FileIO.md

```diff
@@ -0,0 +1,19 @@
+# Microsoft.VisualBasic.FileIO
+
+``` diff
+ namespace Microsoft.VisualBasic.FileIO {
+     public class MalformedLineException : Exception {
+-        [EditorBrowsableAttribute(EditorBrowsableState.Advanced)]
+-        protected MalformedLineException(SerializationInfo info, StreamingContext context);
++        [EditorBrowsableAttribute(EditorBrowsableState.Never)]
++        [ObsoleteAttribute("This API supports obsolete formatter-based serialization. It should not be called or extended by application code.", DiagnosticId="SYSLIB0051", UrlFormat="https://aka.ms/dotnet-warnings/{0}")]
++        protected MalformedLineException(SerializationInfo info, StreamingContext context);
```

Comment:
In the future, could we render these differently?  Right now it looks like we deleted an API and added an API, when this could instead just show an addition for the two new attributes.

Reply:
I agree the output is ugly. Unfortunately, fixing AsmDiff is very low priority right now. It takes a lot of my time to fix it since the code is difficult to maintain and the project has no tests. I am hoping we can eventually move to using GenAPI instead of Microsoft.Cci for generating these API diffs, that would help a lot. I can't promise anything in the short term, but I'll keep this in mind.

Reply:
For reference, I have a PR out here to fix the attribute problem reported: https://github.com/dotnet/arcade/pull/13641